src/ORM/Association/BelongsToMany.php

```diff
@@ -1183,27 +1184,31 @@ function () use ($sourceEntity, $targetEntities, $primaryValue, $options) {
                 $target = $this->getTarget();
 
                 $foreignKey = (array)$this->getForeignKey();
-                $prefixedForeignKey = array_map([$junction, 'aliasField'], $foreignKey);
-
-                $junctionPrimaryKey = (array)$junction->getPrimaryKey();
                 $assocForeignKey = (array)$junction->getAssociation($target->getAlias())->getForeignKey();
 
+                $prefixedForeignKey = array_map([$junction, 'aliasField'], $foreignKey);
+                $junctionPrimaryKey = (array)$junction->getPrimaryKey();
                 $keys = array_combine($foreignKey, $prefixedForeignKey);
                 foreach (array_merge($assocForeignKey, $junctionPrimaryKey) as $key) {
                     $keys[$key] = $junction->aliasField($key);
                 }
 
-                // Find junction records. We join with the association target so that junction
-                // conditions from `targetConditions()` or the finder work.
-                $existing = $junction->find()
-                    ->innerJoinWith($target->getAlias())
-                    ->where($this->targetConditions())
-                    ->where($this->junctionConditions())
-                    ->where(array_combine($prefixedForeignKey, $primaryValue));
-                [$finder, $finderOptions] = $this->_extractFinder($this->getFinder());
-                if ($finder) {
-                    $existing = $target->callFinder($finder, $existing, $finderOptions);
-                }
+                $existing = $this->_appendJunctionJoin($this->find())
```

Comment:
It's too bad we can't cleanly build queries from finders that exclude non-essential parts like contain.

Reply:
Yeah... I also wanted to try and use a `JOIN` with a subquery, but the ORM can't do that. I'm wondering if it is worth expanding the capabilities around subquery joins and making references to the outer queries easier.